critiquebrainz/frontend/views/profile_apps.py

```diff
@@ -57,10 +65,13 @@ def edit(client_id):
 @profile_apps_bp.route('/<client_id>/delete')
 @login_required
 def delete(client_id):
-    client = OAuthClient.query.get_or_404(client_id)
-    if client.user != current_user:
+    try:
+        application = db_oauth_client.get(client_id)
+    except db_exceptions.NoDataFoundException:
+        abort(404)
```

Comment:
`raise NotFound` and `abort(404)` give pretty much the same result. Let's just stick with `NotFound`. Can also pass a message from the "parent" exception.